### PR TITLE
doc: nrf52840_pca10059: add note about reprogramming

### PR DESCRIPTION
**Pull request by mbolivar**
The location of the reset button and how to use it was a little hard for me to figure out, at least. Add some information about it.

### CODE REVIEW

**Comment by mbolivar** [comment]
@lemrey @dbkinder after spending more time with this board and discussing with @lemrey, I've re-worked this to add more information.

File: boards/arm/nrf52840_pca10059/doc/index.rst [hunk 1]
```diff
@@ -107,82 +107,189 @@ Programming and Debugging
...
-   :goals: build
+Option 1: Using the Built-In Bootloader Only
+--------------------------------------------
+
+The board is factory-programmed with Nordic's bootloader from Nordic's nRF5
+SDK. With this option, you'll use Nordic's `nrfutil`_ program to create
+firmware packages supported by this bootloader and flash them to the
+device. Make sure ``nrfutil`` is installed before proceeding.
+
+#. Reset the board into the Nordic bootloader by pressing the RESET button.
```
Comment (dbkinder) [inline boards/arm/nrf52840_pca10059/doc/index.rst]: Will the board stay in bootloader mode long enough to do the rest of these steps (will it time out)?  Or should this step be moved later in the step list?  It seems like you'd want to compile and package the software, then reset the board right before you flash.  I think.

File: boards/arm/nrf52840_pca10059/doc/index.rst [hunk 2]
```diff
@@ -107,82 +107,189 @@ Programming and Debugging
...
+Option 1 later.
+
+Install `nrfutil`_ and `mcumgr`_ first, and make sure MCUboot's ``imgtool`` is
+available for signing your binary for MCUboot as described on :ref:`west-sign`.
+
+Next, do the **one-time setup** to flash MCUboot. We'll assume you've cloned
+the `MCUboot`_ repository into the directory ``mcuboot``, and that it is next
+to the zephyr repository on your computer.
+
+#. Reset the board into the Nordic bootloader as described above.
```
Comment (dbkinder) [inline boards/arm/nrf52840_pca10059/doc/index.rst]: Same here.  Should you wait to reset the board until after you have the software compiled and packaged?

**Review by dbkinder — COMMENTED** [review]
Nice.   I had a couple of questions about step orders before approving though.

File: boards/arm/nrf52840_pca10059/doc/index.rst (hunk 1 above)
Comment (mbolivar) [reply boards/arm/nrf52840_pca10059/doc/index.rst]: The board stays in bootloader mode forever (unlike `arduino_101` and boards which behave like it).

File: boards/arm/nrf52840_pca10059/doc/index.rst (hunk 2 above)
Comment (mbolivar) [reply boards/arm/nrf52840_pca10059/doc/index.rst]: MCUboot will stay in serial recovery mode forever.

**Comment by mbolivar** [comment]
Questions addressed and a couple of typos fixed.

File: boards/arm/nrf52840_pca10059/doc/index.rst [hunk 3]
```diff
@@ -107,82 +107,189 @@ Programming and Debugging
...
+   **Do not press RESET**; that will run the Nordic bootloader, which is
+   different than MCUboot.
+
+   A serial port will enumerate on your board. On Windows, "MCUBOOT" should
+   appear under "Other Devices" in the Device Manager (in addition to the usual
+   ``COMx`` device). On Linux, something like
+   :file:`/dev/serial/by-id/usb-ZEPHYR_MCUBOOT_0.01-if00` should be created.
 
-Create an application package, using nrfutil::
+   If no serial port appears, try plugging it in again, making sure SW1 is
```
Comment (lemrey) [inline boards/arm/nrf52840_pca10059/doc/index.rst]: The procedure should work always, without the need to be retried; especially the MCUboot step, there should be no need to retry that.

**Review by lemrey — APPROVED** [review]
Thank you @mbolivar , left one small comment.

File: boards/arm/nrf52840_pca10059/doc/index.rst (hunk 3 above)
Comment (mbolivar) [reply boards/arm/nrf52840_pca10059/doc/index.rst]: Sure, the code is definitely working; I'm just putting this here for people like me who didn't press SW1 properly and needed to retry. It's concrete troubleshooting advice for if the desired result doesn't happen.

**Comment by mbolivar** [comment]
recheck

**Comment by mbolivar** [comment]
Not sure what the checkpatch issue is? It's not adding any details as a comment.

**Comment by nashif** [comment]
> Not sure what the checkpatch issue is? It's not adding any details as a comment.

Looks like the .svg file is generating too many errors and exceeding the allowable message size we can post to github.

you can always run checkpatch locally..

**Pull request merged**